src/pkgdev/scripts/pkgdev_tatt.py

```diff
@@ -0,0 +1,261 @@
+    docs="""
+        Include a test run for packages which define ``src_test`` phase
+        (in the ebuild or inherited from eclass).
+    """)
+use_opts.add_argument(
+    '-u', '--use-combos', default=0, type=arghparse.positive_int, metavar='NUMBER',
+    help='maximal number USE combinations to be tested')
+use_opts.add_argument(
+    '--ignore-prefixes', default=[], action=arghparse.CommaSeparatedValuesAppend,
+    help='USE flags prefixes that won\'t be randomized',
+    docs="""
+        Comma separated USE flags prefixes that won't be randomized. This is
+        useful for USE flags such as ``python_targets_``. Note that this
+        doesn't affect preference, but because of specific REQUIRED_USE will
+        still be changed from defaults.
+    """)
+random_use_opts = use_opts.add_mutually_exclusive_group()
+random_use_opts.add_argument(
+    '--use-default', dest='random_use', const='d', action='store_const',
+    help='Prefer to use default use flags configuration')
+random_use_opts.add_argument(
+    '--use-random', dest='random_use', const='r', action='store_const',
+    help='Turn on random use flags, with default USE_EXPAND')
+random_use_opts.add_argument(
+    '--use-expand-random', dest='random_use', const='R', action='store_const',
+    help='Turn on random use flags, including USE_EXPAND')
+
+packages_opts = tatt.add_argument_group('manual packages options')
+packages_opts.add_argument(
+    '-p', '--packages', metavar='TARGET', nargs='+',
+    help='extended atom matching of packages')
+bug_state = packages_opts.add_mutually_exclusive_group()
+bug_state.add_argument(
+    '-s', '--stablereq', dest='keywording', default=None, action='store_false',
+    help='Test packages for stable keywording requests')
+bug_state.add_argument(
+    '-k', '--keywording', dest='keywording', default=None, action='store_true',
+    help='Test packages for keywording requests')
+
+template_opts = tatt.add_argument_group('template options')
+template_opts.add_argument(
+    '--template-file',
+    help='Template file to use for the job script')
+template_opts.add_argument(
+    '--logs-dir', default='~/logs',
+    help='Directory to save build logs for failing tasks')
+template_opts.add_argument(
+    '--emerge-opts', default='',
+    help='Options to be passed to emerge invocations')
+
+accept_keywords = Path('/etc/portage/package.accept_keywords')
+
+@tatt.bind_final_check
+def _validate_args(parser, namespace):
+    if namespace.bug is not None:
+        if namespace.keywording is not None:
+            parser.error('cannot use --bug with --keywording or --stablereq')
+        if namespace.packages:
+            parser.error('cannot use --bug with --packages')
+    elif not namespace.packages:
+        parser.error('no action requested, use --bug or --packages')
+
+    if not namespace.test and not namespace.use_combos:
+        parser.error('no action requested, use --test or --use-combos')
+
+    if namespace.random_use is None:
```

Comment:
It's been a while, but you can use `.set_default(random_use='r')` to handle the mux mutating a common attribute.

See https://docs.python.org/3/library/argparse.html#argument-default for other alternatives.